src/App.jsx

```diff
@@ -14,33 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_NAME = 'name';
+const SORT_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function preparedGoodsData(goods, sortValue, reversed) {
+  const preparedData = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortValue) {
+    preparedData.sort((good1, good2) => {
+      switch (sortValue) {
+        case SORT_BY_NAME:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedData.reverse();
+  }
+
+  return preparedData;
+}
+
+export const App = () => {
+  const [sortValue, setSortValue] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = preparedGoodsData(goodsFromServer, sortValue, reversed);
+
+  const sortState = sortValue || reversed;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  function isReversed() {
```

Comment:
This function name is a bit confusion, cuz u r not checking if this is true or false but setting new value. I`d change it.

Reply:
> This function name is a bit confusion, cuz u r not checking if this is true or false but setting new value. I`d change it.

Good point, should be `handleReverse`